### PR TITLE
Add solutions to Exercise6Test.java. Closes #113

### LINKED ISSUE #113: Add solutions for Exercise6Test in company-kata-solutions
[Exercise6Test](https://github.com/eclipse/eclipse-collections-kata/blob/master/company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java)

### PR DESCRIPTION
**Pull request by nsardana-bny**
Signed-off-by: nsardana-bny <neha.sardana@bnymellon.com>

Closes #113

### CODE REVIEW

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java [hunk 1]
```diff
@@ -62,7 +67,9 @@ public void sortedTotalOrderValue()
     @Test
     public void sortedTotalOrderValueUsingPrimitives()
     {
-        MutableDoubleList sortedTotalValues = null;
+        MutableDoubleList sortedTotalValues = ListIterate
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java:70]: Hi @nsardana-bny , this should call the method directly on `this.company.getCustomers()` instead.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java [hunk 2]
```diff
@@ -126,6 +145,11 @@ public void supplierNamesAsTildeDelimitedString()
     @Test
     public void deliverOrdersToLondon()
     {
+        this.company.getCustomers()
+                .asLazy()
+                .selectWith((Predicate2<Customer, String>) (customer, city) -> customer.getCity().equals(city),"London")
+                .flatCollect(Customer::getOrders)
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java:151]: Hi @nsardana-bny, please create a method on `Customer` if it doesn't exist named `livesIn:`. Then the call to `selectWith` can use a method reference. The cast to `Predicate2` should also be unnecessary.

Comment (nsardana-bny) [reply company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java:152]: Hi @donraab , thanks for your comments. I have made the requested changes. Please review and let me know. Thanks.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java [hunk 3]
```diff
@@ -126,6 +146,11 @@ public void supplierNamesAsTildeDelimitedString()
     @Test
     public void deliverOrdersToLondon()
     {
+        this.company.getCustomers()
+                .asLazy()
+                .selectWith(Customer::livesIn,"London")
+                .flatCollect(Customer::getOrders)
+                .each(Procedures.cast(Order::deliver));
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java:153]: The `Procedures.cast` should be unnecessary here. Please try and remove. Thanks!

Comment (nsardana-bny) [reply company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise6Test.java]: Hi Don. Just removed Procedures.cast now. I see it is not required here. Thanks for the tip!

**Pull request merged**